src/components/internal/CopyText/CopyText.tsx

```diff
@@ -1,89 +1,106 @@
 import cx from 'classnames';
-import { useCallback, useState } from 'preact/hooks';
+import { HTMLProps } from 'preact/compat';
+import { TranslationKey } from '../../../translations';
+import { useCallback, useMemo, useState } from 'preact/hooks';
 import useCoreContext from '../../../core/Context/useCoreContext';
+import Icon from '../Icon';
 import Button from '../Button';
 import { ButtonVariant } from '../Button/types';
-import Icon from '../Icon';
 import { Tooltip } from '../Tooltip/Tooltip';
 import './CopyText.scss';
-import { HTMLProps } from 'preact/compat';
 
 type CopyTextProps = {
-    textToCopy: string;
+    buttonLabelKey?: TranslationKey;
     isHovered?: boolean;
-    buttonLabel?: string;
     showCopyTextTooltip?: boolean;
     type?: 'Link' | 'Text' | 'Default';
+    textToCopy: string;
+    visibleText?: string;
 } & HTMLProps<HTMLSpanElement>;
 
-const CopyText = ({ textToCopy, isHovered, buttonLabel, showCopyTextTooltip = true, type = 'Link', ...restProps }: CopyTextProps) => {
-    const { i18n } = useCoreContext();
+const BASE_CLASSNAME = 'adyen-pe-copy-text';
+
+const classes = {
+    base: BASE_CLASSNAME,
+    container: BASE_CLASSNAME + '__container',
+    containerInformation: BASE_CLASSNAME + '__container--information',
+    icon: BASE_CLASSNAME + '__icon',
+    iconInformation: BASE_CLASSNAME + '__icon--information',
+    information: BASE_CLASSNAME + '__information',
+    label: BASE_CLASSNAME + '__label',
+    text: BASE_CLASSNAME + '__text',
+};
 
-    const [tooltipLabel, setTooltipLabel] = useState(i18n.get('copy'));
+const CopyText = ({ buttonLabelKey, isHovered, textToCopy, visibleText, showCopyTextTooltip = true, type = 'Link', ...restProps }: CopyTextProps) => {
+    const { i18n } = useCoreContext();
+    const [isCopied, setIsCopied] = useState(false);
+    const resetIsCopied = useCallback(() => setIsCopied(false), []);
+    const copyButtonLabel = useMemo(() => i18n.get(buttonLabelKey ?? 'copy'), [i18n, buttonLabelKey]);
```

Comment:
Same as the previous command, what do you think about renaming copyButtonLabel to indicate it is only used for aria-label

```suggestion
    const ariaLabel = useMemo(() => i18n.get(buttonLabelKey ?? 'copy'), [i18n, buttonLabelKey]);
```